experts-migration-commands.php

```diff
@@ -0,0 +1,742 @@
+			https://youtu.be/'.$id.'
+			</div></figure>
+			<!-- /wp:embed --> ';
+						}
+					}					
+				}
+	
+				$formatted_content = sprintf( $gutenberg_template, 
+					$profile_pic_url, 
+					$name, 
+					$positions, 
+					$faculty_links,
+					$department_list, 
+					$campus_links,
+					$field_links,
+					$expertise,
+					$pronouns,
+					$pronunciation,
+					implode('', $email_list),
+					implode('', $phone_list),
+					implode('', $website_list),
+					implode('', $social_media_list),
+					$media_contact,
+					$languages,
+					$news_feed,
+					$media_gallery,
+					$embeds
+				);
+	
+				$maybe_update =  get_post_meta( $post_id, 'related_post', true );
+				$id = ( $maybe_update ? $maybe_update : '' );
+	
+				$excerpt = "<h3>$positions</h3>$expertise</p>";
+				$expert_post = array(
+					'ID'			=> $id,
+					'post_title'	=> $name,
+					'post_content'  => $formatted_content,
+					'post_status'   => 'publish',
+					'post_excerpt'		=> $excerpt
+	
+				);
+				$new_post_id = wp_insert_post( $expert_post, false );
+				
+				// save post meta 			
+				if ( $profile_meta["clone_pronunciation"]["textarea"] ) {
+					update_post_meta( $new_post_id, 'self_identification', $profile_meta["clone_self_identification"]["textarea"] );
+				}
+				if ( $profile_meta["clone_internal_memo"]["textarea"] ) {
+					update_post_meta( $new_post_id, 'internal_memo', $profile_meta["clone_internal_memo"]["textarea"] );
+				}
+				update_post_meta( $new_post_id, '_thumbnail_id', $profile_pic );
+	
+				update_post_meta( $post_id, 'related_post', $new_post_id );
+	
+				$this->ubc_experts_insert_new_terms($faculties, $new_post_id, 'category');
+				$this->ubc_experts_insert_new_terms($campuses, $new_post_id, 'category');
+				$this->ubc_experts_insert_new_terms($fields, $new_post_id, 'profile_field');
+				// remove uncategorized
+				wp_remove_object_terms($new_post_id, 1, 'category');
+			}
+		}
+		
+		$paged++;
+
+		if ( $paged <= $profiles->max_num_pages ) {
+			$this->process_experts(array($paged), array());
+		}
+	}
+	
+	public function process_fields_terms (){
+	
+		// taxonomies
+		$faculties = get_terms( 'profile_cct_faculty' );
+		$campuses = get_terms( 'profile_cct_campus' );
+		$fields = get_terms( 'profile_cct_field' );
+		if ( !empty( $fields ) ) {
+			foreach( $fields as $field ){
+	
+				if ( $field->parent ) {
+	
+					// get parent term
+					$get_parent = get_term_by( 'term_id', $field->parent, 'profile_cct_field' );
+					// get parent field
+					$get_parent_field = get_term_by('slug', $get_parent->slug, 'profile_field');
+	
+					// check if parent exists.
+					if ( is_wp_error( $get_parent_field ) || $get_parent_field =='' ) {
+						// insert part term
+						$wp_parent_term = wp_insert_term( $get_parent->name, 'profile_field', array( 'slug' => $get_parent->slug ) );
```

Comment:
Can we validate and sanitize this data before inserting please? Same for the next few `wp_insert_term()` calls.